packages/database/testing/testing.module.ts

```diff
@@ -48,98 +49,75 @@ export class DatabaseTestingModule implements OnModuleDestroy {
-                replicaSet: server.opts.replSet.name,
-                poolSize: 200
-              }
-            );
-
-            const checkStatus = async () => {
-              const status = await client
-                .db()
-                .admin()
-                .replSetGetStatus();
-
-              const hasUnreadyMembers = status.members.some(member => member.state == 5);
-              return hasUnreadyMembers
-                ? Promise.reject("Some of the members didn't become ready within 10s")
-                : Promise.resolve();
-            };
-
-            const wait = (till: number = 2) =>
-              new Promise(resolve => setTimeout(resolve, till * 1000));
+            const connection = await MongoClient.connect(await server.getConnectionString(), {
+              replicaSet: server.opts.replSet.name,
+              poolSize: Number.MAX_SAFE_INTEGER
+            });
 
-            await checkStatus()
-              .catch(() => wait().then(() => checkStatus()))
-              .catch(() => wait(3).then(() => checkStatus()))
-              .catch(() => wait(5).then(() => checkStatus()));
+            const retry = <T>(fn: () => Promise<T>, ms) =>
+              new Promise<T>(resolve => {
+                fn()
+                  .then(resolve)
+                  .catch(() => {
+                    setTimeout(() => retry(fn, ms).then(resolve), ms);
+                  });
+              });
 
-            return client;
+            await retry(
+              () =>
+                connection
+                  .db()
+                  .admin()
+                  .replSetGetStatus()
+                  .then(status => {
+                    const hasPrimary = (status.members as Array<any>).some(
+                      member => member.state == "1" /* PRIMARY */
+                    );
+                    return hasPrimary ? Promise.resolve() : Promise.reject();
+                  }),
+              1000
+            );
+            return connection;
           },
           inject: [MongoMemoryReplSet]
         },
         {
           provide: DatabaseService,
           useFactory: async (client: MongoClient, server: MongoMemoryReplSet) =>
             client.db(await server.getDbName()),
-
           inject: [MongoClient, MongoMemoryReplSet]
         }
       ],
       exports: [DatabaseService, MongoClient]
     };
   }
 
-  static async setDelayedReplica(dbService: DatabaseService) {
-    //get current configure and reconfigure it
-    let config = (await dbService.executeDbAdminCommand({replSetGetConfig: 1})).config;
-    config.version = config.version + 1;
-    config.members[2] = {
-      ...config.members[2],
-      hidden: true,
-      priority: 0,
-      slaveDelay: 5,
-      tags: {slaveDelay: "true"}
-    };
-    await dbService.executeDbAdminCommand({replSetReconfig: config});
-    //wait until replicaSet reconfigured
-    await new Promise(resolve => setTimeout(resolve, 10000));
-  }
```

Comment:
History service needs to this delay settings if I'm not wrong.

Reply:
It didn't fail when I ran the tests so I decided to delete these lines.